File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/stock/OperationStockAdminService.java
package ma.zs.stocky.service.facade.admin.stock;

import java.util.List;
import ma.zs.stocky.bean.core.stock.OperationStock;
import ma.zs.stocky.dao.criteria.core.stock.OperationStockCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface OperationStockAdminService extends  IService<OperationStock,OperationStockCriteria>  {

    List<OperationStock> findByMagasinSourceId(Long id);
    int deleteByMagasinSourceId(Long id);
    long countByMagasinSourceId(Long id);
    List<OperationStock> findByMagasinDestId(Long id);
    int deleteByMagasinDestId(Long id);
    long countByMagasinDestId(Long id);




}

File: frontend/src/app/shared/service/admin/stock/OperationStockAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {OperationStockDto} from 'src/app/shared/model/stock/OperationStock.model';
import {OperationStockCriteria} from 'src/app/shared/criteria/stock/OperationStockCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class OperationStockAdminService extends AbstractService<OperationStockDto, OperationStockCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): OperationStockDto {
        return new OperationStockDto();
    }

    public constrcutCriteria(): OperationStockCriteria {
        return new OperationStockCriteria();
    }

    get API() {
        return environment.apiUrlStockyservice + 'admin/operationStock/';
    }
}
